In [1]:
import numpy as np
import pandas as pd

In [2]:
df= pd.read_csv("Sample_Superstore.csv",  encoding='latin1')

In [3]:
df

Row ID        Order ID  Order Date   Ship Date       Ship Mode  \
0          1  CA-2016-152156   11/8/2016  11/11/2016    Second Class   
1          2  CA-2016-152156   11/8/2016  11/11/2016    Second Class   
2          3  CA-2016-138688   6/12/2016   6/16/2016    Second Class   
3          4  US-2015-108966  10/11/2015  10/18/2015  Standard Class   
4          5  US-2015-108966  10/11/2015  10/18/2015  Standard Class   
...      ...             ...         ...         ...             ...   
9989    9990  CA-2014-110422   1/21/2014   1/23/2014    Second Class   
9990    9991  CA-2017-121258   2/26/2017    3/3/2017  Standard Class   
9991    9992  CA-2017-121258   2/26/2017    3/3/2017  Standard Class   
9992    9993  CA-2017-121258   2/26/2017    3/3/2017  Standard Class   
9993    9994  CA-2017-119914    5/4/2017    5/9/2017    Second Class   

     Customer ID     Customer Name    Segment        Country             City  \
0       CG-12520       Claire Gute   Consumer  United States        Henderson   
1       CG-12520       Claire Gute   Consumer  United States        Henderson   
2       DV-13045   Darrin Van Huff  Corporate  United States      Los Angeles   
3       SO-20335    Sean O'Donnell   Consumer  United States  Fort Lauderdale   
4       SO-20335    Sean O'Donnell   Consumer  United States  Fort Lauderdale   
...          ...               ...        ...            ...              ...   
9989    TB-21400  Tom Boeckenhauer   Consumer  United States            Miami   
9990    DB-13060       Dave Brooks   Consumer  United States       Costa Mesa   
9991    DB-13060       Dave Brooks   Consumer  United States       Costa Mesa   
9992    DB-13060       Dave Brooks   Consumer  United States       Costa Mesa   
9993    CC-12220      Chris Cortes   Consumer  United States      Westminster   

      ... Postal Code  Region       Product ID         Category Sub-Category  \
0     ...       42420   South  FUR-BO-10001798        Furniture    Bookcases   
1     ...       42420   South  FUR-CH-10000454        Furniture       Chairs   
2     ...       90036    West  OFF-LA-10000240  Office Supplies       Labels   
3     ...       33311   South  FUR-TA-10000577        Furniture       Tables   
4     ...       33311   South  OFF-ST-10000760  Office Supplies      Storage   
...   ...         ...     ...              ...              ...          ...   
9989  ...       33180   South  FUR-FU-10001889        Furniture  Furnishings   
9990  ...       92627    West  FUR-FU-10000747        Furniture  Furnishings   
9991  ...       92627    West  TEC-PH-10003645       Technology       Phones   
9992  ...       92627    West  OFF-PA-10004041  Office Supplies        Paper   
9993  ...       92683    West  OFF-AP-10002684  Office Supplies   Appliances   

                                           Product Name     Sales  Quantity  \
0                     Bush Somerset Collection Bookcase  261.9600         2   
1     Hon Deluxe Fabric Upholstered Stacking Chairs,...  731.9400         3   
2     Self-Adhesive Address Labels for Typewriters b...   14.6200         2   
3         Bretford CR4500 Series Slim Rectangular Table  957.5775         5   
4                        Eldon Fold 'N Roll Cart System   22.3680         2   
...                                                 ...       ...       ...   
9989                             Ultra Door Pull Handle   25.2480         3   
9990  Tenex B1-RE Series Chair Mats for Low Pile Car...   91.9600         2   
9991                              Aastra 57i VoIP phone  258.5760         2   
9992  It's Hot Message Books with Stickers, 2 3/4" x 5"   29.6000         4   
9993  Acco 7-Outlet Masterpiece Power Center, Wihtou...  243.1600         2   

      Discount    Profit  
0         0.00   41.9136  
1         0.00  219.5820  
2         0.00    6.8714  
3         0.45 -383.0310  
4         0.20    2.5164  
...        ...       ...  
9989      0.20    4.1028  
9990      0.00   15.6332  
9991      0.20 

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Row ID         9994 non-null   int64  
 1   Order ID       9994 non-null   object 
 2   Order Date     9994 non-null   object 
 3   Ship Date      9994 non-null   object 
 4   Ship Mode      9994 non-null   object 
 5   Customer ID    9994 non-null   object 
 6   Customer Name  9994 non-null   object 
 7   Segment        9994 non-null   object 
 8   Country        9994 non-null   object 
 9   City           9994 non-null   object 
 10  State          9994 non-null   object 
 11  Postal Code    9994 non-null   int64  
 12  Region         9994 non-null   object 
 13  Product ID     9994 non-null   object 
 14  Category       9994 non-null   object 
 15  Sub-Category   9994 non-null   object 
 16  Product Name   9994 non-null   object 
 17  Sales          9994 non-null   float64
 18  Quantity

In [5]:
df.head(10)

Row ID        Order ID  Order Date   Ship Date       Ship Mode Customer ID  \
0       1  CA-2016-152156   11/8/2016  11/11/2016    Second Class    CG-12520   
1       2  CA-2016-152156   11/8/2016  11/11/2016    Second Class    CG-12520   
2       3  CA-2016-138688   6/12/2016   6/16/2016    Second Class    DV-13045   
3       4  US-2015-108966  10/11/2015  10/18/2015  Standard Class    SO-20335   
4       5  US-2015-108966  10/11/2015  10/18/2015  Standard Class    SO-20335   
5       6  CA-2014-115812    6/9/2014   6/14/2014  Standard Class    BH-11710   
6       7  CA-2014-115812    6/9/2014   6/14/2014  Standard Class    BH-11710   
7       8  CA-2014-115812    6/9/2014   6/14/2014  Standard Class    BH-11710   
8       9  CA-2014-115812    6/9/2014   6/14/2014  Standard Class    BH-11710   
9      10  CA-2014-115812    6/9/2014   6/14/2014  Standard Class    BH-11710   

     Customer Name    Segment        Country             City  ...  \
0      Claire Gute   Consumer  United States        Henderson  ...   
1      Claire Gute   Consumer  United States        Henderson  ...   
2  Darrin Van Huff  Corporate  United States      Los Angeles  ...   
3   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   
4   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   
5  Brosina Hoffman   Consumer  United States      Los Angeles  ...   
6  Brosina Hoffman   Consumer  United States      Los Angeles  ...   
7  Brosina Hoffman   Consumer  United States      Los Angeles  ...   
8  Brosina Hoffman   Consumer  United States      Los Angeles  ...   
9  Brosina Hoffman   Consumer  United States      Los Angeles  ...   

  Postal Code  Region       Product ID         Category Sub-Category  \
0       42420   South  FUR-BO-10001798        Furniture    Bookcases   
1       42420   South  FUR-CH-10000454        Furniture       Chairs   
2       90036    West  OFF-LA-10000240  Office Supplies       Labels   
3       33311   South  FUR-TA-10000577        Furniture       Tables   
4       33311   South  OFF-ST-10000760  Office Supplies      Storage   
5       90032    West  FUR-FU-10001487        Furniture  Furnishings   
6       90032    West  OFF-AR-10002833  Office Supplies          Art   
7       90032    West  TEC-PH-10002275       Technology       Phones   
8       90032    West  OFF-BI-10003910  Office Supplies      Binders   
9       90032    West  OFF-AP-10002892  Office Supplies   Appliances   

                                        Product Name     Sales  Quantity  \
0                  Bush Somerset Collection Bookcase  261.9600         2   
1  Hon Deluxe Fabric Upholstered Stacking Chairs,...  731.9400         3   
2  Self-Adhesive Address Labels for Typewriters b...   14.6200         2   
3      Bretford CR4500 Series Slim Rectangular Table  957.5775         5   
4                     Eldon Fold 'N Roll Cart System   22.3680         2   
5  Eldon Expressions Wood and Plastic Desk Access...   48.8600         7   
6                                         Newell 322    7.2800         4   
7                     Mitel 5320 IP Phone VoIP phone  907.1520         6   
8  DXL Angle-View Binders with Locking Rings by S...   18.5040         3   
9                   Belkin F5C206VTEL 6 Outlet Surge  114.9000         5   

   Discount    Profit  
0      0.00   41.9136  
1      0.00  219.5820  
2      0.00    6.8714  
3      0.45 -383.0310  
4      0.20    2.5164  
5      0.00   14.1694  
6      0.00    1.9656  
7      0.20   90.7152  
8      0.20    5.7825  
9      0.00   34.4700  

[10 rows x 21 columns]

In [6]:
df.tail(10)

Row ID        Order ID  Order Date   Ship Date       Ship Mode  \
9984    9985  CA-2015-100251   5/17/2015   5/23/2015  Standard Class   
9985    9986  CA-2015-100251   5/17/2015   5/23/2015  Standard Class   
9986    9987  CA-2016-125794   9/29/2016   10/3/2016  Standard Class   
9987    9988  CA-2017-163629  11/17/2017  11/21/2017  Standard Class   
9988    9989  CA-2017-163629  11/17/2017  11/21/2017  Standard Class   
9989    9990  CA-2014-110422   1/21/2014   1/23/2014    Second Class   
9990    9991  CA-2017-121258   2/26/2017    3/3/2017  Standard Class   
9991    9992  CA-2017-121258   2/26/2017    3/3/2017  Standard Class   
9992    9993  CA-2017-121258   2/26/2017    3/3/2017  Standard Class   
9993    9994  CA-2017-119914    5/4/2017    5/9/2017    Second Class   

     Customer ID     Customer Name    Segment        Country         City  \
9984    DV-13465  Dianna Vittorini   Consumer  United States   Long Beach   
9985    DV-13465  Dianna Vittorini   Consumer  United States   Long Beach   
9986    ML-17410      Maris LaWare   Consumer  United States  Los Angeles   
9987    RA-19885      Ruben Ausman  Corporate  United States       Athens   
9988    RA-19885      Ruben Ausman  Corporate  United States       Athens   
9989    TB-21400  Tom Boeckenhauer   Consumer  United States        Miami   
9990    DB-13060       Dave Brooks   Consumer  United States   Costa Mesa   
9991    DB-13060       Dave Brooks   Consumer  United States   Costa Mesa   
9992    DB-13060       Dave Brooks   Consumer  United States   Costa Mesa   
9993    CC-12220      Chris Cortes   Consumer  United States  Westminster   

      ... Postal Code  Region       Product ID         Category Sub-Category  \
9984  ...       11561    East  OFF-LA-10003766  Office Supplies       Labels   
9985  ...       11561    East  OFF-SU-10000898  Office Supplies     Supplies   
9986  ...       90008    West  TEC-AC-10003399       Technology  Accessories   
9987  ...       30605   South  TEC-AC-10001539       Technology  Accessories   
9988  ...       30605   South  TEC-PH-10004006       Technology       Phones   
9989  ...       33180   South  FUR-FU-10001889        Furniture  Furnishings   
9990  ...       92627    West  FUR-FU-10000747        Furniture  Furnishings   
9991  ...       92627    West  TEC-PH-10003645       Technology       Phones   
9992  ...       92627    West  OFF-PA-10004041  Office Supplies        Paper   
9993  ...       92683    West  OFF-AP-10002684  Office Supplies   Appliances   

                                           Product Name    Sales  Quantity  \
9984                     Self-Adhesive Removable Labels   31.500        10   
9985  Acme Hot Forged Carbon Steel Scissors with Nic...   55.600         4   
9986  Memorex Mini Travel Drive 64 GB USB 2.0 Flash ...   36.240         1   
9987  Logitech G430 Surround Sound Gaming Headset wi...   79.990         1   
9988                    Panasonic KX - TS880B Telephone  206.100         5   
9989                             Ultra Door Pull Handle   25.248         3   
9990  Tenex B1-RE Series Chair Mats for Low Pile Car...   91.960         2   
9991                              Aastra 57i VoIP phone  258.576         2   
9992  It's Hot Message Books with Stickers, 2 3/4" x 5"   29.600         4   
9993  Acco 7-Outlet Masterpiece Power Center, Wihtou...  243.160         2   

      Discount   Profit  
9984       0.0  15.1200  
9985       0.0  16.1240  
9986       0.0  15.2208  
9987       0.0  28.7964  
9988       0.0  55.6470  
9989       0.2   4.1028  
9990       0.0  15.6332  
9991       0.2  19.3932  
9992       0.0  13.3200  
9993       0.0  72.9480  

[10 rows x 21 columns]

In [7]:
df.describe(include="all")

Row ID        Order ID Order Date   Ship Date       Ship Mode  \
count   9994.000000            9994       9994        9994            9994   
unique          NaN            5009       1237        1334               4   
top             NaN  CA-2017-100111   9/5/2016  12/16/2015  Standard Class   
freq            NaN              14         38          35            5968   
mean    4997.500000             NaN        NaN         NaN             NaN   
std     2885.163629             NaN        NaN         NaN             NaN   
min        1.000000             NaN        NaN         NaN             NaN   
25%     2499.250000             NaN        NaN         NaN             NaN   
50%     4997.500000             NaN        NaN         NaN             NaN   
75%     7495.750000             NaN        NaN         NaN             NaN   
max     9994.000000             NaN        NaN         NaN             NaN   

       Customer ID  Customer Name   Segment        Country           City  \
count         9994           9994      9994           9994           9994   
unique         793            793         3              1            531   
top       WB-21850  William Brown  Consumer  United States  New York City   
freq            37             37      5191           9994            915   
mean           NaN            NaN       NaN            NaN            NaN   
std            NaN            NaN       NaN            NaN            NaN   
min            NaN            NaN       NaN            NaN            NaN   
25%            NaN            NaN       NaN            NaN            NaN   
50%            NaN            NaN       NaN            NaN            NaN   
75%            NaN            NaN       NaN            NaN            NaN   
max            NaN            NaN       NaN            NaN            NaN   

        ...   Postal Code  Region       Product ID         Category  \
count   ...   9994.000000    9994             9994             9994   
unique  ...           NaN       4             1862                3   
top     ...           NaN    West  OFF-PA-10001970  Office Supplies   
freq    ...           NaN    3203               19             6026   
mean    ...  55190.379428     NaN              NaN              NaN   
std     ...  32063.693350     NaN              NaN              NaN   
min     ...   1040.000000     NaN              NaN              NaN   
25%     ...  23223.000000     NaN              NaN              NaN   
50%     ...  56430.500000     NaN              NaN              NaN   
75%     ...  90008.000000     NaN              NaN              NaN   
max     ...  99301.000000     NaN              NaN              NaN   

       Sub-Category     Product Name         Sales     Quantity     Discount  \
count          9994             9994   9994.000000  9994.000000  9994.000000   
unique           17             1850           NaN          NaN          NaN   
top         Binders  Staple envelope           NaN          NaN          NaN   
freq           1523               48           NaN          NaN          NaN   
mean            NaN              NaN    229.858001     3.789574     0.156203   
std             NaN              NaN    623.245101     2.225110     0.206452   
min             NaN              NaN      0.444000     1.000000     0.000000   
25%             NaN              NaN     17.280000     2.000000     0.000000   
50%             NaN              NaN     54.490000     3.000000     0.200000   
75%             NaN              NaN    209.940000     5.000000     0.200000   
max             NaN              NaN  22638.480000    14.000000     0.800000   

             Profit  
count   9994.000000  
unique          NaN  
top             NaN  
freq            NaN  
mean      28.656896  
std      234.260108  
min    -6599.978000  
25%        1.728750  
50%        8.666500  
75%       29.364000  
max     8399.976000  

[11 rows x 21 columns]

In [8]:
df.shape

(9994, 21)

In [9]:
df.isnull().sum()
df.duplicated().sum()

0

In [10]:
df.nunique()

Row ID           9994
Order ID         5009
Order Date       1237
Ship Date        1334
Ship Mode           4
Customer ID       793
Customer Name     793
Segment             3
Country             1
City              531
State              49
Postal Code       631
Region              4
Product ID       1862
Category            3
Sub-Category       17
Product Name     1850
Sales            5825
Quantity           14
Discount           12
Profit           7287
dtype: int64

### Data Set Schema Summary


### Metrics 
 Sales 
     min : 0.444000
     mean : 229.858001	
     max : 22638.480000
 Profit 
     min : -6599.978000 (loss)
     mean : 28.656896
     max : 8399.976000
 Quantity
     min : 1.000000	
     mean : 3.789574
     max : 14.000000
 Discount 
     min : 0
     mean : 15.62
     max : 80

### Dimension
 Region : 4 
 Category : 3
 Sub-category : 17